### ISSUE TITLE
GH Packages Publishing is difficult

### ISSUE THREAD

**Issue by SydneyhSmith** [PowerShell/PSResourceGet#1163]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PowerShellGet)
- [X] Search the existing issues.

### Steps to reproduce

For example, one of the bigger pain points is actually publishing to GH Packages because of the required Repository URI, my workaround is to publish to a temporary local disk repository, uncompress the nuspec, directly inject the URI via importing the XML and using XPATH, saving it, then using dotnet nuget to push the recompressed Nuget package up to github. This works, but is far from ideal

from @Adrian-andersson

### Expected behavior

```console
works via PSResourceGet
```

### Actual behavior

```console
Doesn't
```

### Environment data

```powershell
PowerShellGet beta21 is broken for GH so pinned to 19
```

**Comment by alerickson** [comment]
@adrian-andersson publishing to GH Packages worked in beta 19? or was it just searching that worked?

**Comment by anamnavi** [comment]
This will be resolved in our next release, by PR #1224 . To publish to Github Packages you will need either: -Credential where username is your github username and password is your PAT with publish permissions/access, OR, -ApiKey <PAT with publish permissions/access>

We will be updating docs with info/requirements on publishing to the specific repo's.
